pkg/cmd/cli/cmd/set/env.go

```diff
@@ -176,6 +177,20 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 
 	skipped := 0
 	for _, info := range infos {
+		ok := f.UpdateBuildStrategyForObject(info.Object, func(spec *buildapi.BuildStrategy) {
```

Comment:
+1 for `EnvironmentForObject` as a factory method that handles bcs, and objects with podtemplates